storybookjs/storybook
# Fix a bug with buildStaticStandalone resolving too early

Issue: `buildStaticStandalone` returns a `Promise` for its build completing, but that Promise doesn't take into account the actual compiler work, meaning that dependents that want to create static builds have no way of reliably knowing when the build completes.

## What I did

I made a fairly simple change to `build-static.js` that only resolves the `buildStaticStandalone` async result once the build completes or errors.

## How to test

Is this testable with Jest or Chromatic screenshots? No
Does this need a new example in the kitchen sink apps? No
Does this need an update to the documentation? No

You can reproduce this via a standalone static build like this one:

```js
const build = require('@storybook/react/standalone')
const packageJson = require('./package.json')

const configDir = './.storybook'
const outputDir = './storybook-static'

build({
  mode: 'static',
  packageJson,
  configDir,
  outputDir,
  port: 60001,
  host: 'localhost'
}).then(() => {
  console.log('storybook build should be finished', outputDir)
}).catch((err) => {
  console.error('storybook build error', err)
})
```

Note that this is a bugfix for https://github.com/hydrateio/docz-plugin-storybook/pull/4

cc @mergebandit
